### DISCUSSION TITLE
Reading TIF frames.  Is there a way to ignore 'compression not supported' error?

### DISCUSSION THREAD

**Discussion by medianotion** [dlemstra/Magick.NET#1355] [Help]
I have a TIF that has frames that Imagemagik/Magick.Net does not support.  

I want to use Magick.Net to read the TIF and extract the frame bytes to be processed by another library.  However, I can't get Magick.Net to ignore the frame compression/format.  I haven't found any .Net library that will read a TIF and ignore the frame compression/format (bitmircle libtiff, etc)

ChartGPT is saying this should work but it's not an option for ReadSettings:

```
    MagickReadSettings settings = new MagickReadSettings
    {
        ThrowExceptionOnUnsupportedFormat = false // Ignore unsupported compression
    };

```
And this also throws the error:

```
    MagickReadSettings settings = new MagickReadSettings
    {
           IgnoreLayers = true 
    };

```

Please let me know if you have any ideas or suggestions.

Thanks!

**Comment by dlemstra** [comment]
Can you share a file that can be used to reproduce this issue? And `IgnoreLayers` is a property of `TiffReadDefines` and that might work in this situation but you will only get the combined image and not all the frames of the image.

**Comment by medianotion** [comment]
Thank you for responding.

My samples are all private documents that I can't post.  I can't create a sample because I don't have the library that created them.  

If it helps, I have successfully used [tifffile](https://pypi.org/project/tifffile) in python on a OSX to extract the bytes from a frame.  The result was a JP2 file in the frames that could be viewed in OSX Preview.  However, on a Windows 10 or Ubuntu 20.04 the JP2 file comes out as a 0 byte file.  

ImageMagick is reported to support JP2 but for some reason this JP2 is not supported.

My goal is to get the frames out as JP2 using Magick.Net and then find some [open source](https://github.com/cinderblocks/OpenJpegDotNet) library that will convert the JP2 to a standard JPEG.

Example Python that extracts frame bytes on OSX:

```
import tifffile

with tifffile.TiffFile('sample.tif') as tif:
    # open raw file handle
    fh = tif.filehandle
    # iterate the tif pages
    for page in tif.pages:
        # get offset/length pairs
        for index, (offset, bytecount) in enumerate(zip(page.dataoffsets, page.databytecounts)):
            # seek file handle to offset
            _ = fh.seek(offset)
            # read num bytes
            data = fh.read(bytecount)
            bytes = fh.read(page.databytecounts[0])
            # write JP2 bytes to individual file
            with open(f'image{index}.jp2', 'wb') as f:
                f.write(bytes)


```

**Reply by dlemstra** [reply]
Is it possible to share the file with just me through an email?

**Comment by medianotion** [comment]


**Reply by dlemstra** [reply]
You can find my email on my GitHub profile so feel free to reach out to me.
